x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -620,13 +624,22 @@ export function resultsServiceProvider(mlClient: MlClient, client?: IScopedClust
     const finalResults: GetDatafeedResultsChartDataResult = {
       bucketResults: [],
       datafeedResults: [],
+      annotationResultsRect: [],
+      annotationResultsLine: [],
     };
 
     const { getDatafeedByJobId } = datafeedsProvider(client!, mlClient);
-    const datafeedConfig = await getDatafeedByJobId(jobId);
+    const [DATAFEED, JOBS] = [0, 1];
+    const results = await Promise.all([
+      getDatafeedByJobId(jobId),
+      mlClient.getJobs({ job_id: jobId }),
+    ]);
 
-    const { body: jobsResponse } = await mlClient.getJobs({ job_id: jobId });
-    if (jobsResponse.count === 0 || jobsResponse.jobs === undefined) {
+    const datafeedConfig = results[DATAFEED] as estypes.MlDatafeed;
+    const jobsApiResponse = results[JOBS] as ApiResponse<estypes.MlGetJobsResponse> | undefined;
+    const jobsResponse = jobsApiResponse?.body as estypes.MlGetJobsResponse;
+
```

Comment:
```suggestion
    const [datafeedConfig, { body: jobsResponse }] = await Promise.all([
      getDatafeedByJobId(jobId),
      mlClient.getJobs({ job_id: jobId }),
    ]);
```

Reply:
Updated in 3f2c45c807b4eb637c8e3c7e220203c328128f39